packages/graphql/src/translate/create-projection-and-params.ts

```diff
@@ -389,9 +389,15 @@ function createProjectionAndParams({
 
         if (connectionField) {
             if (!inRelationshipProjection) {
-                if (!res.meta.connectionFields) res.meta.connectionFields = [];
-                res.meta.connectionFields.push(field as ResolveTree);
-                res.projection.push(literalElements ? `${field.name}: ${field.name}` : `${field.name}`);
+                if (!res.meta.connectionFields) {
+                    res.meta.connectionFields = [];
+                }
+
+                const f = field as ResolveTree;
```

Comment:
It feels to me like instead of this reassignment, `field` should be of type `ResolveTree` in the reducer arguments instead of `any`.